doc/devel/documenting_mpl.rst

```diff
@@ -352,104 +379,143 @@ See the `~.Axes.hlines` documentation for how this renders.
+   simplification for the type description and explain the type more
+   precisely in the text.
 
-Linking to other code
----------------------
-To link to other methods, classes, or modules in Matplotlib you can encase
-the name to refer to in back ticks, for example:
+   Generally, the `numpydoc docstring guide`_ conventions apply. The following
+   rules expand on them where the numpydoc conventions are not specific.
 
-.. code-block:: python
+   Use ``float`` for a type that can be any number.
 
-  `~matplotlib.collections.LineCollection`
+   Use ``array-like`` for homogeneous numeric sequences, which could
+   typically be a numpy.array. Dimensionality may be specified using ``2D``,
+   ``3D``, ``n-dimensional``. If you need to have variables denoting the
+   sizes of the dimensions, use capital letters in brackets
+   (``array-like (M, N)``). When refering to them in the text they are easier
+   read and no special formatting is needed.
 
-It is also possible to add links to code in Python, Numpy, Scipy, or Pandas.
-Sometimes it is tricky to get external Sphinx linking to work; to check that
-a something exists to link to the following shell command outputs a list of all
-objects that can be referenced (in this case for Numpy)::
+   ``float`` is the implicit default dtype for array-likes. For other dtypes
+   use ``array-like of int``.
 
-  python -m sphinx.ext.intersphinx 'https://docs.scipy.org/doc/numpy/objects.inv'
+   Some possible uses::
 
+      2D array-like
+      array-like (N)
+      array-like (M, N)
+      array-like (M, N, 3)
+      array-like of int
 
-Function arguments
-------------------
-Function arguments and keywords within docstrings should be referred to using
-the ``*emphasis*`` role. This will keep Matplotlib's documentation consistent
-with Python's documentation:
+   Non-numeric homogeneous sequences are described as lists, e.g.::
 
-.. code-block:: rst
+      list of str
+      list of `.Artist`
 
-  Here is a description of *argument*
+Referencing types
+   Generally, the rules from referring-to-other-code_ apply. More specifically:
 
-Do not use the ```default role```:
+   Use full references ```~matplotlib.colors.Normalize``` with an
+   abbreviation tilde in parameter types. While the full name helps the
+   reader of plain text docstrings, the HTML does not need to show the full
+   name as it links to it. Hence, the ``~``-shortening keeps it more readable.
 
+   Use abbreviated links ```.Normalize``` in the text.
 
-.. code-block:: rst
+   .. code-block:: rst
 
-   Do not describe `argument` like this.
+      norm : `~matplotlib.colors.Normalize`, optional
+         A `.Normalize` instance is used to scale luminance data to 0, 1.
 
-nor the ````literal```` role:
+   .. note::
 
-.. code-block:: rst
+      Sphinx claims to automatically link types in ``Parameters`` and
+      ``Returns`` sections. However, this does currently not work reliably.
+      Therefore we still have to enclose them in backticks.
```

Comment:
Is this true? I believe that the only section where backticks are not necessary is the `See Also` section.